sentinelsat/sentinel.py

```diff
@@ -1427,33 +1427,42 @@ def _parse_odata_response(product):
     return output
 
 
-def placename_to_wkt(placename):
-    """Geocodes the placename to rectangular bounding extents using Nominatim API and
+def placename_to_wkt(place_name):
+    """Geocodes the place name to rectangular bounding extents using Nominatim API and
        returns the corresponding 'ENVELOPE' form Well-Known-Text.
 
     Parameters
     ----------
-    placename : str
+    place_name : str
         the query to geocode
 
+    Raises
+    ------
+    ValueError
+        If no matches to the place name were found.
+
     Returns
     -------
-    full name and coordinates of the queried placename
-        list in the form [wkt string in form 'ENVELOPE(minX, maxX, maxY, minY)', string placeinfo]
+    wkt_envelope : str
+        Bounding box of the location as an 'ENVELOPE(minX, maxX, maxY, minY)' WKT string.
+    info : Dict[str, any]
+        Matched location's metadata returned by Nominatim.
     """
-
-    rqst = requests.post(
-        "https://nominatim.openstreetmap.org/search", params={"q": placename, "format": "geojson"}
+    rqst = requests.get(
+        "https://nominatim.openstreetmap.org/search",
+        params={"q": place_name, "format": "geojson"},
+        headers={"User-Agent": "sentinelsat/" + sentinelsat_version},
     )
-    # Check that the response from Openstreetmapserver has status code 2xx
-    # and that the response is valid JSON.
     rqst.raise_for_status()
-    jsonlist = rqst.json()
-    if len(jsonlist["features"]) == 0:
-        raise ValueError('Unable to find a matching location for "{}"'.format(placename))
+    features = rqst.json()["features"]
+    if len(features) == 0:
+        raise ValueError('Unable to find a matching location for "{}"'.format(place_name))
```

Comment:
This would also be triggered if the nominatim endpoint is unavailable, wouldn't it? In that case the error message might be slightly confusing.

Reply:
It should generate a HTTP error from `rqst.raise_for_status()` if Nominatim is not available.